scipy/spatial/ckdtree.pyx

```diff
@@ -18,8 +18,7 @@ from libcpp.algorithm cimport sort
 from libcpp cimport bool
 
 cimport cython
-
-from multiprocessing import cpu_count
+from os import cpu_count
```

Comment:
this is slightly different, os.cpu_count returns an Optional[int] but multiprocessing.cpu_count raises RuntimeError rather than return None

Reply:
I don't know how `cdef int number_of_processors = cpu_count()` handles errors here

Reply:
@andyfaff `cdef int number_of_processors = cpu_count() or -1` ?

Reply:
@larsoner I dismissed your review to draw your attention to this

Reply:
we could write a little wrapper that maintains current behavior:

https://stackoverflow.com/questions/53537230/difference-between-multiprocessing-cpu-count-and-os-cpu-count#answer-53537394

So something like:

```
def _cpu_count():
        num = os.cpu_count()
        if num is None:
            raise NotImplementedError('cannot determine number of cpus')
        else:
            return num
```

Reply:
I'm not familiar with cpython, how does it handle exceptions at c-extension setup time?

Can you push the change?

Reply:
done

Reply:
Thanks!